File: src/main/java/task/java/leetcode75/lvl1/RunningSumOf1dArray.java
package task.java.leetcode75.lvl1;

public class RunningSumOf1dArray {
    public int[] runningSum(int[] nums) {
        int[] result = new int[nums.length];

        for (int i = 0; i < result.length; i++) {
            for (int j = 0; j <= i; j++) {
                result[i] += nums[j];
            }
        }
        return result;
    }
}

File: src/main/kotlin/task/kotlin/leetcode75/lvl1/RunningSumOf1dArray.kt
package task.kotlin.leetcode75.lvl1

// https://leetcode.com/problems/running-sum-of-1d-array
// time: O(n)
// space: O(1)

class RunningSumOf1dArray {
    fun runningSum(nums: IntArray): IntArray {
        if (nums.isEmpty()) return nums

        var sumOfPrev = nums[0]
        for (i in 1 until nums.size) {
            nums[i] += sumOfPrev
            sumOfPrev = nums[i]
        }
        return nums
    }
}